Automattic/wp-calypso
# reduxify: `withFormBase` and `me/profile`

#### Changes proposed in this Pull Request

* Reduxify `withFormBase` HOC which is currently only used for `me/profile`
* Remove `lib/user` and `lib/user-settings` from `withFormBase` and `me/profile`

Note: this shouldn't be merged until #48454 is merged and this PR is rebased upon those changes.

#### Testing instructions

* Go to `/me`
* Change your profile details and make sure the form behaves exactly the same as on prod 
  * The form should be disabled while submitting a request
  * The button to submit your changes should only be enabled when there are changes
* Make sure the Gravatar profile link is correct
* Make sure you're warned if you want to navigate away after you made changes
* Make sure you can safely navigate away after you saved your changes


part of #24162
